pkg/api/validate/dynamic/quota.go

```diff
@@ -133,16 +83,18 @@ func (dv *dynamic) ValidateQuota(ctx context.Context, oc *api.OpenShiftCluster)
 	}
 
 	requiredResources := map[string]int{}
-	err := addRequiredResources(requiredResources, oc.Properties.MasterProfile.VMSize, 3)
-	if err != nil {
-		return err
+	vm, ok := vmFromTypeName(oc.Properties.MasterProfile.VMSize)
+	if !ok {
+		return fmt.Errorf("unexpected node VMSize %s", oc.Properties.MasterProfile.VMSize)
 	}
+	addRequiredResources(requiredResources, vm, 3)
 	//worker node resource calculation
 	for _, w := range oc.Properties.WorkerProfiles {
-		err = addRequiredResources(requiredResources, w.VMSize, w.Count)
-		if err != nil {
-			return err
+		vm, ok := vmFromTypeName(w.VMSize)
+		if !ok {
+			return fmt.Errorf("unexpected node VMSize %s", w.VMSize)
```

Comment:
```suggestion
			return fmt.Errorf("unexpected WorkerProfile VMSize %s", w.VMSize)
```
Also, should we use `unsupported` here?

```suggestion
			return fmt.Errorf("unsupported WorkerProfile VMSize %s", w.VMSize)
```